crates/re_viewer/src/app.rs

```diff
@@ -885,10 +887,9 @@ impl App {
         // being used in.
         for source in sources {
             match &*source {
-                // These source are typically "finite". We want the loading screen so long as data is
-                // coming in.
+                // No need for a welome screen - data is coming soon!
                 SmartChannelSource::File(_) | SmartChannelSource::RrdHttpStream { .. } => {
-                    return true;
+                    return false;
```

Comment:
This is not going to cut it either, I should have been clearer earlier—sorry for that. This *also* breaks the demo workflow: the loading screen is shown initially, but if you then close the recording, we're back to the loading screen—but that time it should be the welcome screen. This is way I originally used `is_connected()`.

I'm not quite sure how to proceed here. That "is_connected" trick might be valid only if this is the only stream. The hole we dug with that whole "when to show welcome screen" thing is only getting deeper now 😓 

Maybe that's something we can revisit later (maybe when my brain fog dissipate), in this case an issue with 0.9 milestone will do.

Reply:
Argh

I think https://github.com/rerun-io/rerun/issues/3125 covers it well